### ISSUE TITLE
Context for Map

### ISSUE THREAD

**Issue by FintanH** [rust-crdt/rust-crdt#66]
It seems strange to call a function like `len` or `is_empty` just to get the `ReadCtx` for a `Map`. I'd like to suggest adding a function `read_ctx` that returns `ReadCtx<(), A>` for this use case.

Also updating the [example](https://github.com/rust-crdt/rust-crdt/blob/master/examples/reset_remove.rs#L8) showing the use of it.

**Comment by davidrusu** [comment]
Yep, makes sense. We have this implemented for the Orswot already: https://github.com/rust-crdt/rust-crdt/blob/master/src/orswot.rs#L274-L280

Would be good to bring this to the Map as well
